modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx

```diff
@@ -214,25 +249,92 @@ const TeamMembersTable = ({
 		);
 	};
 
+	const saveSubscriptionKey = (singleSubscribedKeys) => {
+		singleSubscribedKeys?.forEach(async (singleSubscribeKey) => {
+			try {
+				await provisioningKeys.putSubscriptionInKey(singleSubscribeKey);
+			} catch (error) {
+				console.error('Error:', error);
+			}
+		});
+	};
+
 	return (
 		<>
-			{open && currentUserRemoving !== undefined && (
+			{open && currentUserRemoving !== undefined && !loadingModal && (
 				<RemoveUserModal
+					isSingleSubscribedUser={isSingleSubscribedUser}
 					modalTitle={i18n.translate('remove-user')}
 					observer={observer}
 					onClose={() => onOpenChange(false)}
-					onRemove={() => remove(currentUserRemoving)}
+					onRemove={() => {
+						if (checkedBoxSubscription) {
+							saveSubscriptionKey(singleSubscribedKeys);
+							remove(currentUserRemoving);
+						}
+
+						remove(currentUserRemoving);
+					}}
 					removing={updating}
 				>
 					<p className="my-0 text-neutral-10">
-						<p>
-							<b>Team Member:</b> {currentUserRemoving.name}
+						<p className="font-weight-bold">
+							{`${i18n.translate('team-member-x')} ${
+								currentUserRemoving?.name
+							}`}
 						</p>
 
-						{i18n.translate(
-							'are-you-sure-you-want-to-remove-this-team-member-from-the-project'
+						{!isSingleSubscribedUser.length ? (
+							<>
+								{i18n.translate(
+									'are-you-sure-you-want-to-remove-this-team-member-from-the-project'
+								)}
+							</>
+						) : (
+							<>
+								{i18n.translate(
+									'there-is-at-least-one-activation-key-for-which-this-team-member-is-the-only-one-subscribed-to-be-notified-before-the-activation-key-expires-are-you-sure-you-want-to-remove-this-team-member-and-their-notifications'
+								)}
+							</>
 						)}
 					</p>
+
+					{!!isSingleSubscribedUser.length && (
+						<div className="align-items-center d-flex pt-3">
+							<ClayCheckbox
+								checked={checkedBoxSubscription}
+								onChange={() =>
+									setCheckedBoxSubscription(
+										(checkedBoxSubcription) =>
+											!checkedBoxSubcription
```

Comment:
`checkedBoxSubcription` is spelled incorrectly. it should be `checkedBoxSubscription`. notice the missing "s" in Subscription